### ISSUE TITLE
백엔드랑 연동중 질문

### ISSUE THREAD

**Issue by jin-park0115** [triptalk-4/triptalk-frontend#69]
멘토님 지금 회원가입쪽 연동중인데 회원가입 성공로직은 된거 같아요, 근데 이제 이메일 중복, 닉네임 중복을 다룰때
무조건 백엔드쪽에서 중복체크 엔드포인트가 있어야 하는 걸까요??

@oinochoe

**Comment by jin-park0115** [comment]
아니네여 답장 기다리는 동안 해봤습니다.


` // 회원가입 요청 함수
  const sendSignupData = async (formData: any) => {
    try {
      const res = await axios.post('http://52.79.200.55:8080/api/users/register', formData);
      return res.data;
    } catch (error: any) {
      if (error.response) {
        console.log('gqw: ', error.response);
        return error.response.data;
      }
      console.error('API 요청 실패:', error);
      throw error;
    }
  };

  // 회원가입 버튼
  const handleSignup = async () => {
    if (isFormValid()) {
      try {
        const formData = {
          email: email.value,
          password: password.value,
          name: name.value,
          nickname: nickName.value,
        };

        const res = await sendSignupData(formData);
        console.log(res);
        if (res.registerOk) {
          alert(res.registerOk);
          navigator('/main');
        } else if (res) {
          console.log(res);
          setError(res);
        } else {
          setError(res || '회원가입에 실패했습니다.');
        }
      } catch (error: any) {
        console.error('error:', error);
        setError('서버와 연결이 되지 않습니다.');
      }
    }
  };`
  
  계속 회원가입 버튼 쪽에서 res.error 를 하고있어서 응답을 받지 못했습니다.
  그래서 회원가입 요청 함수 쪽에서 error 안에 있는 response 키 받아서 콘솔을 찍었더니 중복이메일, 중복닉네임 응답이 오는거를 확인해서 return 값으로 error.response.data 받아서 회원가입 버튼 기능에서 res로 error 뜨면 백엔드 쪽에서 주는 응답값 띄웠습니다.  
 지금 잘 해결되었습니다.

**Comment by oinochoe** [comment]
```javascript
// 회원가입 요청 함수
const sendSignupData = async (formData: any) => {
try {
const res = await axios.post('http://52.79.200.55:8080/api/users/register', formData);
return res.data;
} catch (error: any) {
if (error.response) {
console.log('gqw: ', error.response);
return error.response.data;
}
console.error('API 요청 실패:', error);
throw error;
}
};

// 회원가입 버튼
const handleSignup = async () => {
if (isFormValid()) {
try {
const formData = {
email: email.value,
password: password.value,
name: name.value,
nickname: nickName.value,
};

    const res = await sendSignupData(formData);
    console.log(res);
    if (res.registerOk) {
      alert(res.registerOk);
      navigator('/main');
    } else if (res) {
      console.log(res);
      setError(res);
    } else {
      setError(res || '회원가입에 실패했습니다.');
    }
  } catch (error: any) {
    console.error('error:', error);
    setError('서버와 연결이 되지 않습니다.');
  }
}
};
```

**Comment by oinochoe** [comment]
네 error일 때 response안에 어떤 에러코드가 올지에 대한 정의를 더 세분화해서 처리하시면 더 좋습니다.

현업에서는 errorCode를 다 정의해서 리턴해주기 때문에 그 상황마다 처리하는 것들에 대한 메시지 및 처리 내용이 다릅니다. 

참고해보세요!

@leesh2985 @jin-park0115 @Dongyeong98

**Comment by jin-park0115** [comment]
넵!
